shims/common/src/main/scala/org/apache/gluten/GlutenConfig.scala

```diff
@@ -1211,7 +1211,7 @@ object GlutenConfig {
       .internal()
       .doc("The IO threads for connector split preloading")
       .intConf
-      .createWithDefault(0)
+      .createWithDefaultFunction(() => SQLConf.get.getConfString("spark.executor.cores", "1").toInt)
```

Comment:
this thread executor will be used in DirectBufferedInput, not just for split preloading.

Reply:
Updated the comments, thanks.